torchvision/datasets/omniglot.py

```diff
@@ -0,0 +1,207 @@
+from __future__ import print_function
+from PIL import Image
+from functools import reduce
+import os
+import random
+import torch.utils.data as data
+from .utils import download_url, check_integrity, list_dir, list_files
+
+
+class Omniglot(data.Dataset):
+    """`Omniglot <https://github.com/brendenlake/omniglot>`_ Dataset.
+    Args:
+        root (string): Root directory of dataset where directory
+            ``omniglot-py`` exists.
+        background (bool, optional): If True, creates dataset from the "background" set, otherwise
+            creates from the "evaluation" set. This terminology is defined by the authors.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        download (bool, optional): If true, downloads the dataset zip files from the internet and
+            puts it in root directory. If the zip files are already downloaded, they are not
+            downloaded again.
+        force_extract (bool, optional): If true, extracts the downloaded zip file irrespective
+            of the existence of an extracted folder with the same name
+    """
+    folder = 'omniglot-py'
+    download_url_prefix = 'https://github.com/brendenlake/omniglot/raw/master/python'
+    zips_md5 = [
+        ['images_background', '68d2efa1b9178cc56df9314c21c6e718'],
+        ['images_evaluation', '6b91aef0f799c5bb55b94e3f2daec811'],
+        # Provision in future
+        # ['images_background_small1', 'e704a628b5459e08445c13499850abc4'],
```

Comment:
Please can you remove the commented out bits and open an issue to add these once this is merged?